internal/controllers/clusterextension_controller.go

```diff
@@ -267,6 +273,65 @@ func mapBDStatusToInstalledCondition(existingTypedBundleDeployment *rukpakv1alph
 	}
 }
 
+// setDeprecationStatus will set the appropriate deprecation statuses for a ClusterExtension
+// based on the provided bundle
+func setDeprecationStatus(ext *v1alpha1.ClusterExtension, bundle *catalogmetadata.Bundle) {
+	// unset conditions
+	meta.RemoveStatusCondition(&ext.Status.Conditions, v1alpha1.TypePackageDeprecated)
+	meta.RemoveStatusCondition(&ext.Status.Conditions, v1alpha1.TypeChannelDeprecated)
+	meta.RemoveStatusCondition(&ext.Status.Conditions, v1alpha1.TypeBundleDeprecated)
+	meta.RemoveStatusCondition(&ext.Status.Conditions, v1alpha1.TypeDeprecated)
```

Comment:
Is it more appropriate to set the conditions to false and set necessary fields to their empty values?

Reply:
Unit tests fail when the conditions are removed. made it so that these conditions are always present after a reconciliation